lib/common/xml.c

```diff
@@ -1319,7 +1337,8 @@ pcmk__xml_write_fd(const xmlNode *xml, const char *filename, int fd,
         return errno;
     }
 
-    return write_xml_stream(xml, filename, stream, compress, nbytes);
+    return write_xml_stream(xml, pcmk__s(filename, "unnamed file"), stream,
```

Comment:
You'd have to do this at all the callers, or within write_xml_stream()

Reply:
It is done at all the callers. There are only two, and the other one (`pcmk__xml_write_file()`) CRM_CHECKs `filename`. `write_xml_stream()` is just a static helper for those two.